.github/workflows/on-push.yml

```diff
@@ -0,0 +1,54 @@
+name: On Push
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "main" ]
+
+jobs:
+  build-all:
+    secrets: inherit
+    strategy:
+      fail-fast: false
+      matrix:
+        test-group: [
+          { arch: grayskull },
+          { arch: wormhole_b0 },
+          { arch: blackhole },
+        ]
+    uses: ./.github/workflows/build-source.yml
+    with:
+      arch: ${{ matrix.test-group.arch }}
+
+  build-tests:
+    secrets: inherit
+    strategy:
+      fail-fast: false
+      matrix:
+        test-group: [
+          # { arch: grayskull },
+          { arch: wormhole_b0 },
+          # { arch: blackhole },
+        ]
+    uses: ./.github/workflows/build-tests.yml
+    with:
+      arch: ${{ matrix.test-group.arch }}
+
+  test-all:
+    secrets: inherit
+    needs: build-tests
+    strategy:
+      fail-fast: false
+      matrix:
+        test-group: [
+          # { arch: grayskull, card: e75 },
```

Comment:
Either add a comment why these lines are under comment or open the issue to add grayskull tests when grayskull machines are available

Reply:
Added